android/tangram/src/com/mapzen/tangram/MapController.java

```diff
@@ -738,12 +760,27 @@ public void setDebugFlag(DebugFlag flag, boolean on) {
 
     /**
      * Enqueue a scene component update with its corresponding YAML node value
-     * @param componentPath The YAML component path delimited by a '.' (example "scene.animated")
-     * @param value A YAML valid string (example "{ property: true }" or "true")
+     * @param sceneUpdate A {@code SceneUpdate}
      */
-    public void queueSceneUpdate(String componentPath, String value) {
+    public void queueSceneUpdate(SceneUpdate sceneUpdate) {
         checkPointer(mapPointer);
-        nativeQueueSceneUpdate(mapPointer, componentPath, value);
+        nativeQueueSceneUpdate(mapPointer, sceneUpdate.getPath(), sceneUpdate.getValue());
+    }
+
+    /**
+     * Enqueue a scene component update with its corresponding YAML node value
+     * @param sceneUpdates List of {@code SceneUpdate}
+     */
+    public void queueSceneUpdate(List<SceneUpdate> sceneUpdates) {
+        checkPointer(mapPointer);
```

Comment:
No the code won't break but it could be helpful to the developer if we provided some explicit feedback rather than failing silently in the case of invalid input.

Also passing data across the JNI is expensive. If we can avoid that by checking the input in Java first that would be preferable.